nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.ts

```diff
@@ -1,8 +1,301 @@
+    );
+  }
+
+  ngOnInit(): void {
+    this.initiateJSClient();
+    this.elasticSearchReindexService.pageTitle.next(
+      `${ELASTIC_SEARCH_LABELS.NXQL_QUERY_REINDEX_TITLE}`
+    );
+    this.nxqlQueryHintSanitized = this.sanitizer.bypassSecurityTrustHtml(
+      ELASTIC_SEARCH_LABELS.NXQL_INPUT_HINT
+    );
+
+    this.nxqlReindexingLaunchedSubscription =
+      this.nxqlReindexingLaunched$.subscribe((data) => {
+        if (data?.commandId) {
+          this.showReindexLaunchedModal(data?.commandId);
+        }
+      });
+
+    this.nxqlReindexingErrorSubscription = this.nxqlReindexingError$.subscribe(
+      (error) => {
+        if (error) {
+          this.showReindexErrorModal(error);
+        }
+      }
+    );
+  }
+
+  showReindexLaunchedModal(commandId: string | null): void {
+    this.launchedDialogRef = this.dialogService.open(
+      ElasticSearchReindexModalComponent,
+      {
+        disableClose: true,
+        height: ELASTIC_SEARCH_REINDEX_MODAL_DIMENSIONS.height,
+        width: ELASTIC_SEARCH_REINDEX_MODAL_DIMENSIONS.width,
+        data: {
+          type: ELASTIC_SEARCH_LABELS.MODAL_TYPE.launched,
+          title: `${ELASTIC_SEARCH_LABELS.REINDEX_LAUNCHED_MODAL_TITLE}`,
+          launchedMessage: `${ELASTIC_SEARCH_LABELS.REINDEX_LAUNCHED} ${commandId}. ${ELASTIC_SEARCH_LABELS.COPY_MONITORING_ID}`,
+          closeLabel: `${ELASTIC_SEARCH_LABELS.CLOSE_LABEL}`,
+          commandId,
+          copyActionId: `${ELASTIC_SEARCH_LABELS.COPY_ACTION_ID_BUTTON_LABEL}`,
+          isLaunchedModal: true,
+        },
+      }
+    );
+
+    this.launchedDialogClosedSubscription = this.launchedDialogRef
+      .afterClosed()
+      .subscribe((data) => {
+        if (data?.isClosed) {
+          this.onReindexLaunchedModalClose();
+        }
+      });
+  }
+
+  onReindexLaunchedModalClose(): void {
+    this.nxqlReindexForm?.reset();
+    document.getElementById("nxqlQuery")?.focus();
+  }
+
+  showReindexErrorModal(error: HttpErrorResponse): void {
+    this.errorDialogRef = this.dialogService.open(
+      ElasticSearchReindexModalComponent,
+      {
+        disableClose: true,
+        height: ELASTIC_SEARCH_REINDEX_MODAL_DIMENSIONS.height,
+        width: ELASTIC_SEARCH_REINDEX_MODAL_DIMENSIONS.width,
+        data: {
+          type: ELASTIC_SEARCH_LABELS.MODAL_TYPE.error,
+          title: `${ELASTIC_SEARCH_LABELS.REINDEX_ERRROR_MODAL_TITLE}`,
+          errorMessageHeader: `${ELASTIC_SEARCH_LABELS.REINDEXING_ERROR}`,
+          error: error,
+          closeLabel: `${ELASTIC_SEARCH_LABELS.CLOSE_LABEL}`,
+          isErrorModal: true,
+        },
+      }
+    );
+    this.errorDialogClosedSubscription = this.errorDialogRef
+      .afterClosed()
+      .subscribe((data) => {
+        if (data?.isClosed) {
+          this.onReindexErrorModalClose();
+        }
+      });
+  }
+
+  onReindexErrorModalClose(): void {
+    document.getElementById("nxqlQuery")?.focus();
+  }
+
+  initiateJSClient(): void {
```

Comment:
This needs to move to a separate service. Either remove this code for now if this is being added in a separate PR, or rectify it in this PR